python/ray/data/_internal/progress_bar.py

```diff
@@ -113,6 +113,10 @@ def update(self, i: int) -> None:
                 self._bar.total = self._progress
             self._bar.update(i)
 
+    def update_total(self, total: int) -> None:
+        if self._bar and total != self._bar.total:
+            self._bar.total = total
```

Comment:
So we're going to be making this method a general state update (like `_Bar.update`)? Since we're combining 2 completely separate calls into one, there'll be places where we do `update(i=1, total=None)` and also `update(i=None, total=estimated_total)`

Reply:
I was thinking that there would be no `update(i=None, total=estimated_total)`. Is it right? As we'll only update the total when we have made some progress, which will also update `i`. 
If there are other cases where you think we'll only update the total. Having 2 separate methods makes more sense. 
but I don't have strong preference though, it's fine to separate them, just don't forgot to add refresh.

Reply:
oh yeah thats true, I can just combine the calls